libyul/backends/evm/ControlFlowGraphBuilder.cpp

```diff
@@ -341,6 +373,47 @@ void ControlFlowGraphBuilder::operator()(Switch const& _switch)
 {
 	yulAssert(m_currentBlock, "");
 	shared_ptr<DebugData const> preSwitchDebugData = debugDataOf(_switch);
+	auto switchBlock = m_currentBlock;
+	std::cout << "CFGB: Visiting switch expression" << std::endl;
+	StackSlot switchExpr = std::visit(*this, *_switch.expression);
+	CFG::BasicBlock& afterSwitch = m_graph.makeBlock(preSwitchDebugData);
+
+	optional<Case const*> defaultCase;
+	for (Case const& _case: _switch.cases)
+	{
+		if (!_case.value)
+			defaultCase = &_case;
+	}
+	CFG::BasicBlock* defaultCaseBlock = nullptr;
+	if (defaultCase.has_value())
+	{
+		std::cout << "Visiting default case" << std::endl;
+		defaultCaseBlock = &m_graph.makeBlock(debugDataOf(defaultCase.value()->body));
+		m_currentBlock = defaultCaseBlock;
+		(*this)(defaultCase.value()->body);
+	}
+
+	map<u256, CFG::BasicBlock*> cases;
+	for (Case const& _case: _switch.cases)
+	{
+		if (_case.value)
+		{
+			u256 caseVal = valueOfLiteral(*_case.value);
+			cases[caseVal] = &m_graph.makeBlock(debugDataOf(_case.body));
+	std::cout << "Visiting case value " << caseVal << std::endl;
+			m_currentBlock = cases[caseVal];
+			(*this)(_case.body);
+		}
+	}
+
+	m_currentBlock = switchBlock;
+	makeSwitch(debugDataOf(_switch), switchExpr, defaultCaseBlock,
+		cases, afterSwitch);
```

Comment:
In case you have a default case, the switch itself never jumps to ``afterSwitch``, so I think ``afterSwitch`` should only be conditionally in a switch just in case that there is no default case.